petastorm/transform.py

```diff
@@ -34,10 +34,18 @@ def __init__(self, func=None, edit_fields=None, removed_fields=None):
         :param edit_fields: Optional. A list of 4-tuples with the following fields:
           ``(name, numpy_dtype, shape, is_nullable)``
         :param removed_fields: Optional[list]. A list of field names that will be removed from the original schema.
+        :param selected_fields: Optional[list]. A list of field names specify the fields to be selected.
+          If selected_fields specified, The reader schema will preserve the field order in selected_fields.
+
+        Note: For param `removed_fields` and `selected_fields`, user can only specify one of them.
         """
         self.func = func
         self.edit_fields = edit_fields or []
         self.removed_fields = removed_fields or []
+        self.selected_fields = selected_fields or []
+
+        if self.removed_fields and self.selected_fields:
```

Comment:
If `removed_fields` overlaps with `edit_fields`, the original behavior here is to edit the common fields (ignore `removed_fields`). We can also keep the behavior consistent for `selected_fields`: just don't raise an error when `removed_fields` is also provided.

Reply:
Yes. I thought this. But, I think force user providing only one of them make the semantic simpler and leading less error.